test/FunctionalTests/E2ETestBase.cs

```diff
@@ -141,5 +142,19 @@ protected virtual void AssertResponse(HttpResponseMessage response, string respo
         {
             throw new NotImplementedException("Must be overriden in derived types.");
         }
+
+        private static int GetNextPort()
```

Comment:
I assume this code is copied from some other place? Is it worth putting it in one of our test helper classes?